### PR TITLE
Create basic ORY OAuth2 server doc

### PR DESCRIPTION
**Pull request by tomekpapiernik**
**Description**

Changes proposed in this pull request:

- Updated component list - `ory` is enabled in all deployments of Kyma
- Added basic doc to /security - describing the ORY stack, indicating the early stage of development

**Related issue(s)**
#4796

### CODE REVIEW

File: docs/security/03-06-oauth2-server.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  1. Export the required client application details and your cluster domain as environment variables. Run:
+    ```
+    export CLIENT_ID={YOUR_CLIENT_ID}
+    export CLIENT_SECRET={YOUR_CLIENT_SECRET}
+    export DOMAIN={YOUR_CLUSTER_DOMAIN}
+    ```
+
+  2. Send a CURL request to the `/clients` endpoint of the OAuth2 server to register your client with read and write scopes. Run:
+    ```
```
Comment (mmitoraj) [inline docs/security/03-06-oauth2-server.md]: The code block does not render correctly. The command shows in the same line the text is. Adding an additional empty line could help.

File: docs/security/03-06-oauth2-server.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    curl -ik -X POST "https://oauth2-admin.$DOMAIN/clients" -d '{"grant_types":["client_credentials"], "client_id":"'$CLIENT_ID'", "client_secret":"'$CLIENT_SECRET'", "scope":"read write"}'
+    ```
+
+## See the OAuth2 server in action
```
Comment (mmitoraj) [inline docs/security/03-06-oauth2-server.md]:
```suggestion
## OAuth2 server in action
```
Just a suggestion to make the title shorter.

File: docs/security/03-06-oauth2-server.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+After you register an OAuth2 client, go to [this](https://github.com/kyma-incubator/examples/tree/master/ory-hydra/scenarios/client-credentials) Kyma Incubator repository to try a Client Credentials Grant example that showcases the integration.
```
Comment (mmitoraj) [inline docs/security/03-06-oauth2-server.md]: What kind of integration do you mean?

File: docs/security/03-06-oauth2-server.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+You can also interact with the OAuth2 server using its REST API. Read the ORY [documentation](https://www.ory.sh/docs/hydra/sdk/api) to learn more about the available endpoints.
```
Comment (mmitoraj) [inline docs/security/03-06-oauth2-server.md]:
```suggestion
You can also interact with the OAuth2 server using its REST API. Read the official [ORY documentation](https://www.ory.sh/docs/hydra/sdk/api) to learn more about the available endpoints.
```
This is how I usually see this kind of phrase in `kyma-project.io/docs/`.

File: docs/kyma/04-01-overview.md [hunk 5]
```diff
@@ -28,6 +28,7 @@ To make the local development process easier, we introduced the **Kyma Lite** co
 | `service-catalog` | ✅ | ✅ |
 | `service-catalog-addons` | ✅ | ✅ |
 | `nats-streaming` | ✅ | ✅ |
+| `ory` | ✅ | ✅ |
```
[the remark this answers is not in the document]
Comment (tomekpapiernik) [reply docs/kyma/04-01-overview.md]: I don't think there's any rules to that :) Whenever someone adds something new to the list the add it at the end. When I started working on this PR `ory` was last, but when I rebased it turned out these knative components were added after ory.

File: docs/security/03-06-oauth2-server.md (hunk 4 above)
Comment (tomekpapiernik) [reply docs/security/03-06-oauth2-server.md]: 👍

File: docs/security/03-06-oauth2-server.md (hunk 2 above)
Comment (tomekpapiernik) [reply docs/security/03-06-oauth2-server.md]: 👍

File: docs/security/03-06-oauth2-server.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+---
+title: OAuth2 and OpenID Connect server
+type: Details
+---
+
+By default, every Kyma deployment comes with an OAuth2 authorization server solution from [ORY](https://www.ory.sh/). The `ory` [component](https://github.com/kyma-project/kyma/tree/master/resources/ory) consists of three elements:
+
+- [Hydra](https://github.com/ory/hydra) OAuth2 and OpenID Connect server which issues access, refresh, and ID tokens to registered clients which call services in a Kyma cluster.
+- [Oathkeeper](https://github.com/ory/oathkeeper) authorization & authentication proxy which authenticates and authorizes incoming requests basing on the list of defined [Access Rules](https://www.ory.sh/docs/oathkeeper/api-access-rules).
+- [Oathkeeper Maester](https://github.com/ory/oathkeeper-maester) Kubernetes controller which feeds Access Rules to the Oathkeeper proxy by creating or updating the Oathkeeper ConfigMap and populating it with rules found in instances of the `rules.oathkeeper.ory.sh/v1alpha1` custom resource.
```
Comment (mmitoraj) [inline docs/security/03-06-oauth2-server.md]:
```suggestion
- [Oathkeeper Maester](https://github.com/ory/oathkeeper-maester) Kubernetes controller which feeds Access Rules to the Oathkeeper proxy by creating or updating the Oathkeeper ConfigMap, and populating it with rules found in instances of the `rules.oathkeeper.ory.sh/v1alpha1` custom resource.
```

File: docs/security/03-06-oauth2-server.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+To interact with the Kyma OAuth2 server, you must register an OAuth2 client. Follow these steps:
+
+1. Export the required client application details and your cluster domain as environment variables. Run:
+  ```
+  export CLIENT_ID={YOUR_CLIENT_ID}
+  export CLIENT_SECRET={YOUR_CLIENT_SECRET}
+  export DOMAIN={YOUR_CLUSTER_DOMAIN}
+  ```
+
+2. Send a CURL request to the `/clients` endpoint of the OAuth2 server to register your client with read and write scopes. Run:
```
Comment (mmitoraj) [inline docs/security/03-06-oauth2-server.md]:
```suggestion
2. Send a CURL request to the `/clients` endpoint of the OAuth2 server to register your client with the read and write scopes. Run:
```

**Pull request merged**